src/main/java/com/kakaotech/team14backend/auth/PrincipalDetails.java

```diff
@@ -13,71 +14,76 @@
 import java.util.Map;
 
 @Data
-public class PrincipalDetails implements UserDetails,OAuth2User {
-
-    private Member member;
-    private Map<String,Object> attributes;
-    public PrincipalDetails(Member member,Map<String,Object> attributes){
-        this.member = member;
-        this.attributes = attributes;
-    }
-    public PrincipalDetails(Member member){
-        this.member = member;
-    }
-
-
-    @Override
-    public Map<String, Object> getAttributes() {
-        return attributes;
-    }
-
-    @Override
-    public Collection<? extends GrantedAuthority> getAuthorities() {
-        Collection<GrantedAuthority> collect = new ArrayList<>();
-        collect.add(new GrantedAuthority() {
-            @Override
-            public String getAuthority() {
-                return String.valueOf(member.getRole());
-            }
-        });
-
-        return collect;
-    }
-
-    @Override
-    public String getPassword() {
-        return null;
-    }
-
-    @Override
-    public String getUsername() {
-        return member.getUserName();
-    }
-
-    @Override
-    public boolean isAccountNonExpired() {
-        return false;
-    }
-
-    @Override
-    public boolean isAccountNonLocked() {
-        return false;
-    }
-
-    @Override
-    public boolean isCredentialsNonExpired() {
-        return false;
-    }
-
-    @Override
-    public boolean isEnabled() {
-        return false;
-    }
-
-    @Override
-    public String getName() {
-        return member.getUserName();
-    }
-
-
+@Getter
+public class PrincipalDetails implements UserDetails, OAuth2User {
+
+  private Member member;
+  private Map<String, Object> attributes;
+
+  public PrincipalDetails(Member member, Map<String, Object> attributes) {
+    this.member = member;
+    this.attributes = attributes;
+  }
+
+  public PrincipalDetails(Member member) {
+    this.member = member;
```

Comment:
```suggestion
    this(member, Map.of());
```
이렇게 하면 어떨까요? attributes에 의존하는 로직을 쓸 때 NPE도 막을 수 있고요.

Reply:
오.. 이런방법이 있군요 감사합니다!!